src/core/term.rs

```diff
@@ -1,44 +1,80 @@
+// Rust Simplicity Library
+// Written in 2020 by
+//   Andrew Poelstra <apoelstra@blockstream.com>
+//   Sanket Kanjalkar <sanket1729@gmail.com>
+//
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the CC0 Public Domain Dedication
+// along with this software.
+// If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+//
+
+//! # Simplicity terms
+//!
+//! Terms are the atomic building blocks of linear Simplicity programs.
+//! A program is a sequence of terms, and
+//! the meaning of a term depends on its program.
+//! We speak of _nodes_ to refer to terms in context of their program.
+//!
+//! This linear representation is meant for (de)serialization and execution of programs.
+//! To compose programs by hand, use the DAG representation of [`super::term_dag`].
+
 use crate::core::iter::DagLike;
 use crate::jet::{Application, JetNode};
 use crate::merkle::cmr::Cmr;
 use std::collections::HashMap;
 use std::hash::Hash;
 
-/// Single, untyped Simplicity node.
-/// May include Bitcoin/Elements extensions.
-///
-/// If Bitcoin/Elements support is not compiled (see `bitcoin` and
-/// `elements` features), then programs using these extensions will
-/// fail to parse.
-///
-/// A node consists of a combinator and its payload
-/// _(references to other nodes, witness data, etc.)_.
-/// A list of nodes forms an untyped Simplicity program,
-/// which represents an untyped Simplicity DAG.
+/// Simplicity term for some witness type and application.
+/// Terms in context of programs are called _nodes_,
+/// and terms have no meaning outside a program.
 ///
-/// References to other nodes are relative indices in the context of a program.
-/// For example, node `InjL(2)` at index 7 represents DAG `InjL → x`,
-/// where `x` is the DAG that the node at index 5 represents.
-/// A node has no meaning without a program.
-///
-/// The node representation is later used for executing Simplicity programs.
+/// Nodes refer to other nodes via indices that are relative to the node's own index.
+/// This simulates a DAG-like structure.
```

Comment:
In 024f9115bc3f626199ebc464c6cf507bf96449fd:

This is not a simulation, nor is it "DAG-like". It's literally a DAG.